Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -0,0 +1,1053 @@
+            {
+                return HTMLInputElement::create(HTMLNames::inputTag, document, /* form */ nullptr, /* createdByParser */ true);
+            }
+        };
+
+        struct Li : ContainerTag<HTMLLIElement, PermittedParents::Special> {
+            static constexpr const char tagName[] = "li";
+        };
+
+        struct Label : ContainsPhrasingContentTag<HTMLLabelElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "label";
+        };
+
+        struct Option : ContainerTag<HTMLOptionElement, PermittedParents::Special> {
+            static constexpr const char tagName[] = "option";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                // <option> can only contain a text content.
+                return self.didFail(HTMLFastPathResult::FailedOptionWithChild, nullptr);
+            }
+        };
+
+        struct Ol : ContainerTag<HTMLOListElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "ol";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Li>();
+            }
+        };
+
+        struct P : ContainsPhrasingContentTag<HTMLParagraphElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "p";
+        };
+
+        struct Select : ContainerTag<HTMLSelectElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "select";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Option>();
+            }
+        };
+
+        struct Span : ContainsPhrasingContentTag<HTMLSpanElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "span";
+        };
+
+        struct Strong : ContainsPhrasingContentTag<HTMLElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "strong";
+
+            static Ref<HTMLElement> create(Document& document)
+            {
+                return HTMLElement::create(HTMLNames::strongTag, document);
+            }
+        };
+
+        struct Ul : ContainerTag<HTMLUListElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "ul";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Li>();
+            }
+        };
+    };
+
+    template<class ParentTag> void parseCompleteInput()
+    {
+        parseChildren<ParentTag>(m_fragment);
+        if (m_position != m_end)
+            didFail(HTMLFastPathResult::FailedDidntReachEndOfInput);
+    }
+
+    // Match ASCII Whitespace according to https://infra.spec.whatwg.org/#ascii-whitespace
+    bool isWhitespace(Char c)
```

Comment:
This looks like a slower version of `isHTMLSpace`, so we could use the real one.

Reply:
OK, I hope our version really is faster.